### PR TITLE
rls: migrate data types to AutoValue

### PR DESCRIPTION
**Pull request by dapengzhang0**
Refactor to use `@AutoValue` for data types. This reduces human mistakes on `equals()`, `hashCode()`, and `toString()` while we are constantly adding and changing member fields of the data type.

### CODE REVIEW

File: rls/src/main/java/io/grpc/rls/RlsRequestFactory.java [hunk 1]
```diff
@@ -73,12 +73,12 @@ RouteLookupRequest create(String service, String method, Metadata metadata) {
       grpcKeyBuilder = keyBuilderTable.get("/" + service + "/*");
     }
     if (grpcKeyBuilder == null) {
-      return new RouteLookupRequest(ImmutableMap.<String, String>of());
+      return RouteLookupRequest.create(ImmutableMap.<String, String>of());
```
Comment (sergiitk) [inline rls/src/main/java/io/grpc/rls/RlsRequestFactory.java:76]: The type seems to be inferable now 
```suggestion
      return RouteLookupRequest.create(ImmutableMap.of());
```

**Review by sergiitk — APPROVED** [review]
LGTM

File: rls/src/main/java/io/grpc/rls/RlsRequestFactory.java (hunk 1 above)
Comment (dapengzhang0) [reply rls/src/main/java/io/grpc/rls/RlsRequestFactory.java:76]: This is due to the recent drop of Java 7 support. We will do a systematic clean up to java8-tify the codebase after a release.

Comment (sergiitk) [reply rls/src/main/java/io/grpc/rls/RlsRequestFactory.java:76]: Got it, didn't know.

**Pull request merged**